### PR TITLE
[MM-61123] Add diagnostics data to the Support Packet

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
When a Support Packet gets generated, the plugin now includes diagnostics data via a `diagnostics.yaml` file into the Packet.

Example content of `focalboard/diagnostics.yaml`:

```yaml
version: 9.0.1
total_boards: 1
active_cards: 8
```

Requires https://github.com/mattermost/mattermost/pull/28833 on the server side.

See https://mattermost.atlassian.net/wiki/spaces/CLOUD/pages/3083108354/Diagnostics+data+from+Core+Plugins+to+the+Support+Packet for more context.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61123

### CODE REVIEW

File: server/services/telemetry/mocks/ServerIface.go
Comment (hanzei) [inline server/services/telemetry/mocks/ServerIface.go:1]: Not used any longer

File: server/boards/support_packet.go
Comment (hanzei) [inline server/boards/support_packet.go:1]: Please let me know if this is the right place for the file.

**Review by harshilsharma63 — APPROVED** [review]

**Comment by harshilsharma63** [comment]
cc @Rajat-Dabade

File: server/boards/support_packet.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+type SupportPacket struct {
+	Version string `yaml:"version"`
+	// The total number of active boards.
+	ActiveBoards int64 `yaml:"active_boards"`
+	// The total number of active cards.
+	ActiveCards int64 `yaml:"active_cards"`
+}
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply server/boards/support_packet.go:23]: Good point. Adding fields for total boards and cards (including archived) would give a better sense of scale and could provide more insight into usage patterns.

File: server/boards/support_packet.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	var result *multierror.Error
+
+	boardCount, err := b.server.Store().GetBoardCount()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get boards count")
+	}
+	usedCardsCount, err := b.server.Store().GetUsedCardsCount()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get used cards count")
+	}
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply server/boards/support_packet.go]: Yup, that's correct. `GetBoardCount` return the board which are not archived (`delete_at` is 0), same goes for `GetUsedCardCount` as well.

File: server/boards/support_packet.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply server/boards/support_packet.go:27]: Updated the PR

**Review by Rajat-Dabade — APPROVED** [review]
LGTM 🚀

**Review by wiggin77 — APPROVED** [review]

**Comment by hanzei** [comment]
Per discussion with @raghavaggarwal2308, QA testing will happen post merge.

**Pull request merged**